Quarkus guide (#1603)
platformsh/platformsh-docs: docs/src/guides/quarkus/deploy.md

---
title: "How to Deploy Quarkus on Platform.sh"
sidebarTitle: "Get started"
weight: -110
layout: single
toc: false
description: |
    Create a Platform.sh account, download a few tools, and prepare to deploy Quarkus.
---

> Quarkus is, in its own words, a cloud-native, (Linux) container-first framework for writing Java applications. It has become popular lately because of its fast boot time and low RSS memory. In this series of articles about Quarkus, we'll discuss how to deploy a Quarkus application even faster to the cloud with Platform.sh.

Reviewer: What is RSS memory?  For people that don't know Java, it means "Really Simple Syndication". :smile:

Author: @Crell RSS = Heap size + MetaSpace + OffHeap size
But, I'll remove this information.

Quarkus became so popular because it is light and straightforward to use. Quarkus has integration with several databases such as [PostgreSQL](configuration/services/postgresql.md), [MariaDB](configuration/services/mysql.md), [MySQL](configuration/services/mysql.md), [MongoDB](configuration/services/mongodb.md), [Redis](configuration/services/redis.md), [Elasticsearch](configuration/services/elasticsearch.md), etc. 

> To Dependency injection, it provides a Quarkus DI based on CDI. As Java Developer, you can let the frameworks handle the Object life cycle, besides taking advantage of the inversion of control (IoC) programming principle.

Reviewer: Not relevant for the Guide.  We assume the reader is already familiar with Quarkus.

{{< note >}}
[Quarkus has a vast guide](https://quarkus.io/guides/), where you can take advantage and learn the complete resources from there.
{{< /note >}}

## Hello World Quarkus

> Let's start with the basics, where we'll create a Quarkus application from scratch and then move it to Platform.sh. There two ways to make a Quarkus application:

Reviewer: We try to avoid "Let's start" and "we" in the documentation.  "We" refers to Platform.sh, not the reader.  The reader is "you".

3. [A file to define routes](configuration/routes/_index.md)

### Application container: `.platform.app.yaml`

The application file is the file where you define how you'll package and run your Quarkus application.

```yaml
name: app
type: "java:11"
disk: 1024
hooks:
    build: ./mvnw package -DskipTests -Dquarkus.package.uber-jar=true
web:
    commands:
        start: java -jar $JAVA_OPTS $CREDENTIAL -Dquarkus.http.port=$PORT target/file.jar
```

Simply explaining the file line by line, we have the following settings.

1. `name`: The application name
2. `type` where you'll define the language, in this case, Java, and the version.
3. `disk`: the space in the disk that the application needs in megabytes.
4. `book.build`: the command to package the application
5. `web.commands`: The order to start the application, it is essential to overwrite the port with `-Dquarkus.http.port=$PORT`

To simplify the application file, we'll use [Shell variables](https://docs.platform.sh/development/variables.html#shell-variables) int the  `.environment` file. That is the right choice because you don't need to change the application file, only the environment file.

> ```she

Reviewer: ```suggestion
```shell
```

---
title: "How to Deploy Quarkus on Platform.sh"
sidebarTitle: "Get started"
weight: -110
layout: single
toc: false
description: |
    Create a Platform.sh account, download a few tools, and prepare to deploy Quarkus.
---

> Quarkus is, in its own words, a cloud-native, (Linux) container-first framework for writing Java applications. It has become popular lately because of the amazingly fast boot time, incredibly low RSS memory. In this series of articles about Quarkus, we'll discuss how to deploy a Quarkus application even faster to the cloud with Platform.sh.

Reviewer 2: ```suggestion
Quarkus is, in its own words, a cloud-native, (Linux) container-first framework for writing Java applications. It has become popular lately because of its fast boot time and low RSS memory. In this series of articles about Quarkus, we'll discuss how to deploy a Quarkus application even faster to the cloud with Platform.sh.
```

Quarkus became so popular because it is light and straightforward to use. Quarkus has integration with several databases such as [PostgreSQL](configuration/services/postgresql.md), [MariaDB](configuration/services/mysql.md), [MySQL](configuration/services/mysql.md), [MongoDB](configuration/services/mongodb.md), [Redis](configuration/services/redis.md), [Elasticsearch](configuration/services/elasticsearch.md), etc. 
To Dependency injection, it provides a Quarkus DI based on CDI. As Java Developer, you can let the frameworks handle the Object life cycle, besides taking advantage of the inversion of control (IoC) programming principle.

> ```

Reviewer 2: ```suggestion

```

> {{< note title="Tip">}}

Reviewer 2: ```suggestion
{{< note >}}
```

[Quarkus has a vast guide](https://quarkus.io/guides/), where you can take advantage and learn the complete resources from there.
{{< /note >}}

> ```

Reviewer 2: ```suggestion

```

## Hello World Quarkus

Let's start with the basics, where we'll create a Quarkus application from scratch and then move it to Platform.sh. There two ways to make a Quarkus application:

1. [Maven Archetype](https://quarkus.io/guides/getting-started): where you can create your first project from a terminal using Maven. E.g.:

>    * ```

Reviewer 2: ```suggestion
    ```
```

     mvn io.quarkus:quarkus-maven-plugin:1.9.2.Final:create \
         -DprojectGroupId=org.acme \
         -DprojectArtifactId=getting-started \
         -DclassName="org.acme.getting.started.GreetingResource" \
         -Dpath="/hello"
     ```

     

2. [The start code link](https://code.quarkus.io/): A Ui interface that helps you bootstrap your Quarkus application and discover its extension ecosystem.

As soon as you generate the application it will create an  `ExemploResource` with the Hello world message.

```java
import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;
@Path("/hello")
public class ExampleResource {
    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public String hello() {
        return "Hello World From Platform.sh";
    }

}
```

> {{< note title="Tip">}}

Reviewer 2: ```suggestion
{{< note >}}
```
